gms/gossiper.cc

```diff
@@ -775,6 +730,84 @@ future<gossiper::endpoint_permit> gossiper::lock_endpoint(inet_address ep) {
     });
 }
 
+
+future<> gossiper::failure_detector_thread_for_node(gms::inet_address node) {
+    auto last = gossiper::clk::now();
+    auto max_duration = std::chrono::milliseconds(_cfg.failure_detector_timeout_in_ms());
+    auto diff = std::chrono::milliseconds(0);
+    auto nr = _live_endpoints.size();
+    while (is_enabled()) {
+        bool failed = false;
+        try {
+            logger.debug("failure_detector_thread: Send echo to node {}", node);
+            co_await _messaging.send_gossip_echo(netw::msg_addr(node), max_duration);
+            logger.debug("failure_detector_thread: Send echo to node {}, status = ok", node);
+        } catch (...) {
+            failed = true;
+            logger.info("failure_detector_thread: Send echo to node {}, status = failed: {}", node, std::current_exception());
+        }
+        auto now = gossiper::clk::now();
+        diff = now - last;
+        if (!failed) {
+            last = now;
```

Comment:
You can move that into the end of the try clause.

Reply:
We need to update last after we calculate the diff. If we move last = now to try clause, we need to call diff = now - last and now() in both the try and except cause. I think current version is simpler.